index.ts

```diff
@@ -628,19 +628,26 @@ commander
  * Opens the script editor in the user's browser.
  */
 commander
-  .command('open')
+  .command('open [scriptId]')
   .description('Open a script')
-  .action((scriptId: string) => {
-    getProjectSettings().then(async ({ scriptId }: ProjectSettings) => {
-      if (scriptId) {
-        if (scriptId.length < 30) {
-          logError(null, ERROR.SCRIPT_ID_INCORRECT(scriptId));
-        } else {
-          console.log(LOG.OPEN_PROJECT(scriptId));
-          open(getScriptURL(scriptId));
-        }
+  .action(async (scriptId: string) => {
+    const openScript = (scriptId) => {
+      if (scriptId.length < 30) {
+        logError(null, ERROR.SCRIPT_ID_INCORRECT(scriptId));
+      } else {
+        console.log(LOG.OPEN_PROJECT(scriptId));
+        open(getScriptURL(scriptId));
       }
-    });
+    };
+    if (scriptId) {
+      openScript(scriptId);
+    } else {
+      getProjectSettings().then(async ({ scriptId }: ProjectSettings) => {
```

Comment:
Let's write this:

```ts
getProjectSettings().then(openScript);
```

```ts
const openScript = (scriptId?) => {
  if (!scriptId) return;
  // if (scriptId.length < 30) { ... length check
  // } else { ... log and open
}
```

In the future (when `getProjectSettings` is async):

```ts
const openScript = (scriptId?) => { ... }
openScript(scriptId ? scriptId : (await getProjectSettings()).scriptId);
```

Reply:
#146 Created issue for async getProjectSettings